rubygem/lib/zeus/load_tracking.rb

```diff
@@ -1,8 +1,15 @@
 module Zeus
   class LoadTracking
     class << self
-      def features_loaded_by(&block)
-        old_features = all_features
+      def add_feature(file)
+        full_path = File.expand_path(file)
+        return unless File.exist?(full_path) && @feature_pipe
+        notify_features([full_path])
```

Comment:
`add_feature` now sends feature to the Zeus master synchronously rather than buffering them until the end of a `features_loaded_by` block. IIRC, this costs 50-100ms over thousands of files. Combined with https://github.com/burke/zeus/pull/579, this release will still be faster at tracking features. We can always add a batched `add_features` later if it becomes a concern.